### PR TITLE
Methods placeholder generator

### PR DESCRIPTION
**Pull request by DonNicoJs**
@mikeu this is what we discussed in #23 I changed a bit the name of the utility because I think even if is a bit longer like this is more expressive which is btw what we are trying to do!

Let me know WDYT! 

Ps: I think we need to fix the linters, I will look into that

### CODE REVIEW

**Review by mikeu — COMMENTED** [review]
I like this approach, I think it has the potential to be more useful than my initial suggestion of a single placeholder used over and over, especially if the minor change I proposed seems reasonable, to make the console warning more informative.

Two related comments on the general approach. First, based on the [Vue reactivity docs](https://v3.vuejs.org/guide/reactivity-fundamentals.html#destructuring-reactive-state) I wonder whether it would be better to return `toRefs` of the current reactive object. This should then allow components to declare something like:
```js
schematics = {
  someLocalNonLeafletMethod,
  ...generateMethodsPlaceholder(["leafletMethod"]),
}
```
Unless your idea was that the `schematics` object should be reserved exclusively for those that must be replaced with dynamically imported Leaflet methods?

Second, I definitely agree that the more expressive name is worth the extra characters involved, but curious what you think of calling it `generatePlaceholderMethods` instead. I think it makes it a little more obvious that multiple methods will be returned, especially if we do make the change to allow destructuring them, but since the function does return a single object with multiple methods on it (and _not_ an array of methods, which some people might assume from the plural?), the current name isn't incorrect either.

File: src/utils.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { watch } from "vue";
+import { watch, reactive } from "vue";
```
Comment (mikeu) [inline src/utils.js:1]:
```suggestion
import { watch, reactive, toRefs } from "vue";
```

File: src/utils.js [hunk 2]
```diff
@@ -70,3 +70,12 @@ export const resetWebpackIcon = (Icon) => {
...
   });
 };
+
+export const generateMethodsPlaceholder = (methods) => {
+  const base = reactive({});
+  return methods.reduce((acc, curr) => {
+    acc[curr] = () =>
+      console.warn("This method has been invoked without being replaced ");
+    return acc;
+  }, base);
```
Comment (mikeu) [inline src/utils.js:80]:
```suggestion
  return toRefs(methods.reduce((acc, curr) => {
    acc[curr] = () =>
      console.warn("This method has been invoked without being replaced ");
    return acc;
  }, base));
```

**Review by mikeu — DISMISSED** [review]
Interested to hear your thoughts on my suggestions and question @DonNicoJs , but whatever you decide, this LGTM!

File: src/utils.js [hunk 3]
```diff
@@ -70,3 +70,12 @@ export const resetWebpackIcon = (Icon) => {
     shadowUrl: require("leaflet/dist/images/marker-shadow.png"),
   });
 };
+
+export const generateMethodsPlaceholder = (methods) => {
+  const base = reactive({});
+  return methods.reduce((acc, curr) => {
```
Comment (mikeu) [inline src/utils.js:76]:
```suggestion
  return toRefs(methods.reduce((acc, curr) => {
```

File: src/utils.js (hunk 2 above)
Comment (mikeu) [inline src/utils.js:80]:
```suggestion
  }, base));
```

File: src/utils.js [hunk 4]
```diff
@@ -70,3 +70,12 @@ export const resetWebpackIcon = (Icon) => {
...
+    acc[curr] = () =>
+      console.warn("This method has been invoked without being replaced ");
```
Comment (mikeu) [inline src/utils.js:78]:
```suggestion
      console.warn(`Method ${curr} has been invoked without being replaced`);
```

**Comment by mikeu** [comment]
Sorry for all the minor one-line edits, wanted to keep the change to the console warning message separate from the addition of `toRefs`.

**Comment by DonNicoJs** [comment]
@mikeu So the issue with `toRef` is that then we would be forced to use all the methods with `methodName.value` which i think is not the best and quite confusing!

Thank you for the suggestion on the error warn! I'll also update the method name, I think your version is definitively better!

**Comment by DonNicoJs** [comment]
@mikeu The PR is ready for another look!

**Comment by mikeu** [comment]
> So the issue with `toRef` is that then we would be forced to use all the methods with `methodName.value` which i think is not the best and quite confusing!

Great point. I do like the idea of being able to spread what `generatePlaceholderMethods` returns, but not so much that it is worth having to use `.value` on every method, so yep, let's stick with `reactive`.

**Review by mikeu — APPROVED** [review]

**Pull request merged**